src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -603,15 +604,25 @@ export class FolderMatch extends Disposable {
 	}
 
 	matches(): (FileMatch | FolderMatchWithResource)[] {
-		return [...this.fileMatches(), ...this.folderMatches()];
+		return [...this.fileMatchesIterator(), ...this.folderMatchesIterator()];
+	}
+
+	fileMatchesArray(): FileMatch[] {
+		// for testing only; try to use fileMatchesIterator for performance reasons
```

Comment:
Comments like this are for the person calling the function, so use jsdoc

Reply:
You can also create the array in the test code itself from calling the iterator version, I don't see why this has to be in product code

Reply:
there are some tests that get the foldermatches of foldermatches and so on, so it looks cleaner if it's something that can be called directly on the object. I can move it over, though!